staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/conversion_test.go

```diff
@@ -0,0 +1,137 @@
+*/
+
+package webhook
+
+import (
+	"reflect"
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/apis/example"
+	examplev1 "k8s.io/apiserver/pkg/apis/example/v1"
+	appsv1register "k8s.io/kubernetes/pkg/apis/apps/v1"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+)
+
+func initiateScheme() *runtime.Scheme {
+	s := runtime.NewScheme()
+	example.AddToScheme(s)
+	examplev1.AddToScheme(s)
+	extensions.AddToScheme(s)
+	appsv1register.AddToScheme(s)
+	return s
+}
+
+func TestConvertToGVK(t *testing.T) {
+	scheme := initiateScheme()
+	w := GenericAdmissionWebhook{
+		convertor: scheme,
+		creator:   scheme,
+	}
+	table := []struct {
+		obj         runtime.Object
+		gvk         schema.GroupVersionKind
+		expectedObj runtime.Object
+	}{
+		{
+			obj: &example.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "pod1",
+					Labels: map[string]string{
+						"key": "value",
+					},
+				},
+				Spec: example.PodSpec{
+					RestartPolicy: example.RestartPolicy("never"),
+				},
+			},
+			gvk: examplev1.SchemeGroupVersion.WithKind("Pod"),
+			expectedObj: &examplev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "pod1",
+					Labels: map[string]string{
+						"key": "value",
+					},
+				},
+				Spec: examplev1.PodSpec{
+					RestartPolicy: examplev1.RestartPolicy("never"),
+				},
+			},
+		},
+		// TODO: move the replicaset definition to k8s.io/apiserver
+		{
+			obj: &extensions.ReplicaSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "rs1",
+					Labels: map[string]string{
+						"key": "value",
+					},
+				},
+				Spec: extensions.ReplicaSetSpec{
+					Replicas: 1,
+				},
+			},
+			gvk: appsv1.SchemeGroupVersion.WithKind("ReplicaSet"),
+			expectedObj: &appsv1.ReplicaSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "rs1",
+					Labels: map[string]string{
+						"key": "value",
+					},
+				},
+				Spec: appsv1.ReplicaSetSpec{
+					Replicas: func() *int32 { var i int32; i = 1; return &i }(),
+				},
+			},
+		},
+		{
+			obj: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "mygroup.k8s.io/v1",
+					"kind":       "Flunder",
+					"data": map[string]interface{}{
+						"Key": "Value",
+					},
+				},
+			},
+			gvk: schema.GroupVersionKind{Group: "mygroup.k8s.io", Version: "v1", Kind: "Flunder"},
+			expectedObj: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "mygroup.k8s.io/v1",
+					"kind":       "Flunder",
+					"data": map[string]interface{}{
+						"Key": "Value",
+					},
+				},
+			},
+		},
+	}
+
+	for _, test := range table {
+		actual, err := w.convertToGVK(test.obj, test.gvk)
+		if err != nil {
+			t.Error(err)
+		}
+		if !reflect.DeepEqual(actual, test.expectedObj) {
+			t.Errorf("\nexpected:\n%#v\ngot:\n %#v\n", test.expectedObj, actual)
+		}
+	}
+}
```

Comment:
I think what I had in mind with my test comment: a test which checks that serialization of the webhook payload correctly serialized the Unstructured object.